### ISSUE TITLE
SQL Ingestion fails with UnicodeDecodeError

### ISSUE THREAD

**Issue by sachinwadhwa** [datahub-project/datahub#7380]
SQL Ingestion fails with Unicode error when using ODBC with MS SQL server, this only happening to one database only. URI_ARGs are:

******
       uri_args:
          driver: "ODBC Driver 18 for SQL Server"
          encrypt: "yes"
          TrustServerCertificate: "Yes"
          ssl: "True"
          ApplicationIntent: "ReadOnly"

We install following libraries:
    apt install -y unixodbc
    apt install -y msodbcsql18 
    pip install 'acryl-datahub[mssql]' pyodbc

Stack Trace:
**********
[2023-02-20 13:31:51,987] INFO     {datahub.cli.ingest_cli:163} - DataHub CLI version: 0.10.0.1
[2023-02-20 13:31:52,007] INFO     {datahub.ingestion.run.pipeline:184} - Sink configured successfully. DataHubRestEmitter: configured to talk to http://datahub-gms:8080 with token: eyJh**********jvxk
[2023-02-20 13:31:54,450] ERROR    {datahub.entrypoints:188} - Command failed: Failed to configure the source (mssql): 'utf-16-le' codec can't decode byte 0x79 in position 72: truncated data
Traceback (most recent call last):
  File "/opt/bitnami/python/lib/python3.11/site-packages/datahub/ingestion/run/pipeline.py", line 115, in _add_init_error_context
    yield
  File "/opt/bitnami/python/lib/python3.11/site-packages/datahub/ingestion/run/pipeline.py", line 197, in __init__
    self.source = source_class.create(
                  ^^^^^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/datahub/ingestion/source/sql/mssql.py", line 195, in create
    return cls(config, ctx)
           ^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/datahub/ingestion/source/sql/mssql.py", line 131, in __init__
    self._populate_column_descriptions(conn, db_name)
  File "/opt/bitnami/python/lib/python3.11/site-packages/datahub/ingestion/source/sql/mssql.py", line 187, in _populate_column_descriptions
    for row in column_metadata:
  File "/opt/bitnami/python/lib/python3.11/site-packages/sqlalchemy/engine/result.py", line 382, in iterrows
    for row in self._fetchiter_impl():
  File "/opt/bitnami/python/lib/python3.11/site-packages/sqlalchemy/engine/cursor.py", line 1810, in _fetchiter_impl
    row = fetchone(self, self.cursor)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/sqlalchemy/engine/cursor.py", line 974, in fetchone
    self.handle_exception(result, dbapi_cursor, e)
  File "/opt/bitnami/python/lib/python3.11/site-packages/sqlalchemy/engine/cursor.py", line 955, in handle_exception
    result.connection._handle_dbapi_exception(
  File "/opt/bitnami/python/lib/python3.11/site-packages/sqlalchemy/engine/base.py", line 2128, in _handle_dbapi_exception
    util.raise_(exc_info[1], with_traceback=exc_info[2])
  File "/opt/bitnami/python/lib/python3.11/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/opt/bitnami/python/lib/python3.11/site-packages/sqlalchemy/engine/cursor.py", line 969, in fetchone
    row = dbapi_cursor.fetchone()
          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/datahub/ingestion/source/sql/mssql.py", line 136, in handle_sql_variant_as_string
    return value.decode("utf-16le")
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/encodings/utf_16_le.py", line 16, in decode
    return codecs.utf_16_le_decode(input, errors, True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'utf-16-le' codec can't decode byte 0x79 in position 72: truncated data

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/bitnami/python/lib/python3.11/site-packages/datahub/entrypoints.py", line 175, in main
    sys.exit(datahub(standalone_mode=False, **kwargs))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/datahub/telemetry/telemetry.py", line 379, in wrapper
    raise e
  File "/opt/bitnami/python/lib/python3.11/site-packages/datahub/telemetry/telemetry.py", line 334, in wrapper
    res = func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/datahub/utilities/memory_leak_detector.py", line 95, in wrapper
    return func(ctx, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/datahub/cli/ingest_cli.py", line 177, in run
    pipeline = Pipeline.create(
               ^^^^^^^^^^^^^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/datahub/ingestion/run/pipeline.py", line 308, in create
    return cls(
           ^^^^
  File "/opt/bitnami/python/lib/python3.11/site-packages/datahub/ingestion/run/pipeline.py", line 196, in __init__
    with _add_init_error_context(f"configure the source ({source_type})"):
  File "/opt/bitnami/python/lib/python3.11/contextlib.py", line 155, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/opt/bitnami/python/lib/python3.11/site-packages/datahub/ingestion/run/pipeline.py", line 117, in _add_init_error_context
    raise PipelineInitError(f"Failed to {step}: {e}") from e
datahub.ingestion.run.pipeline.PipelineInitError: Failed to configure the source (mssql): 'utf-16-le' codec can't decode byte 0x79 in position 72: truncated data

**Comment by sachinwadhwa** [comment]
Resolution here - https://stackoverflow.com/questions/47645224/unicodedecodeerror-utf-16-le


This is almost certainly caused by a known issue with cursor.columns() when using the Access ODBC driver if the table metadata includes "Descriptions" for the columns. The following GitHub issue includes a discussion of the problem and possible workarounds:

https://github.com/mkleehammer/pyodbc/issues/328

TL;DR: The Access ODBC driver returns trailing garbage bytes under those circumstances and some combinations of the bytes will not be valid UTF-16LE encoded characters. A pyodbc output converter function can be used to intercept the byte string and decode it without throwing an exception. The specific workaround is described here:

https://github.com/mkleehammer/pyodbc/issues/328#issuecomment-419655266
